packages/playwright-core/types/types.d.ts

```diff
@@ -15299,6 +15299,11 @@ export interface Request {
      * Total number of bytes from the start of the HTTP response message until (and including) the double CRLF before the body.
      */
     responseHeadersSize: number;
+
+    /**
+     * Total number of bytes received from the start of the HTTP response including the header and body sizes.
+     */
+    transferSize: number;
```

Comment:
It can be easily computed using existing API, so I don't think we want to complicate the API by adding another field that just a some of the other two. Can you start with filing an issue describing the use cases which are not addressed by current API? We usually use issue tracker to decide if a feature is worth implementing.

Reply:
I did try using the existing API and the values were bit mixed up. The reason why I added this to the API was
1. The value of `transferSize = resHeaderSize + responseBodySize` was not always correct, you can also see from this failed test - https://github.com/microsoft/playwright/actions/runs/3137989206/jobs/5096813728. Maybe it could be a Firefox bug alone. 
2. The API was inconsistent between the `server/network` vs `client/network` which makes it difficult for users relying on their own tracer instead of using HarTracer which uses the other API - https://github.com/microsoft/playwright/blob/2a85806cc0a903f676036af7f5fab603ac45205e/packages/playwright-core/src/server/har/harTracer.ts#L339

Happy to close the PR, if you feel its not worth it.

Reply:
By definition `transferSize = resHeaderSize + responseBodySize` from what you describe it seems that there is a bug and this invariant doesn't hold for Firefox. We should fix the instead of exposing another value which is inconsistent with the other two.

Reply:
The header size is computed manually on the client end if browser doesn't provide exact values. This is the case in Firefox and believe we can improve that by changing the code like this because 2 out of 3 value do come from the browser there:

```diff
diff --git a/packages/playwright-core/src/server/firefox/ffNetworkManager.ts b/packages/playwright-core/src/server/firefox/ffNetworkManager.ts
index 2fdc8850b..164e038b0 100644
--- a/packages/playwright-core/src/server/firefox/ffNetworkManager.ts
+++ b/packages/playwright-core/src/server/firefox/ffNetworkManager.ts
@@ -115,8 +115,6 @@ export class FFNetworkManager {
     });
     // "raw" headers are the same as "provisional" headers in Firefox.
     response.setRawResponseHeaders(null);
-    // Headers size are not available in Firefox.
-    response.setResponseHeadersSize(null);
     this._page._frameManager.requestReceivedResponse(response);
   }
 
@@ -127,6 +125,8 @@ export class FFNetworkManager {
     const response = request.request._existingResponse()!;
     response.setTransferSize(event.transferSize);
     response.setEncodedBodySize(event.encodedBodySize);
+    // Headers size are not available in Firefox.
+    response.setResponseHeadersSize(event.transferSize - event.encodedBodySize);
 
     // Keep redirected requests in the map for future reference as redirectedFrom.
     const isRedirected = response.status() >= 300 && response.status() <= 399;
```

Reply:
Thanks for the details around the FF implementation. 

> The API was inconsistent between the server/network vs client/network which makes it difficult for users relying on their own tracer instead of using HarTracer which uses the other API/ 

Do you think if the above is a valid use case? if not, we can close this PR and fix the Firefox issue separately.